src/netaddress.cpp

```diff
@@ -515,6 +517,29 @@ bool CNetAddr::IsAddrV1Compatible() const
     assert(false);
 }
 
+bool CNetAddr::LimitOutboundByNetgroup() const
+{
+    const auto onlynets = gArgs.GetArgs("-onlynet");
+    if (onlynets.size() != 1) {
+        return true;
+    }
+    switch (m_net) {
+    case NET_UNROUTABLE:
+    case NET_IPV4:
+    case NET_IPV6:
+    case NET_INTERNAL:
+        return true;
+    case NET_ONION:
+    case NET_I2P:
+    case NET_CJDNS:
+        return false;
+    case NET_MAX: // m_net is never and should not be set to NET_MAX
+        assert(false);
+    } // no default case, so the compiler can warn about missing cases
+
+    assert(false);
+}
```

Comment:
I think we should generally avoid adding `assert()` to networking code, unless we're sure that it's really better to crash than to continue, eg because the node is in an inconsistent or corrupt state and is unable to continue running. That doesn't seem to be the case here, so I think we could use `Assume(false)` instead, so that we only get crashes in debug builds and our CI environment.

Reply:
If this assert is hit, it means that somebody has added a new entry in `enum Network`, has forgotten to update this function, has ignored the compiler warning and has ignored the CI failure. Or a memory corruption has occurred and `CNetAddr::m_net` has been overwritten with random bytes. I think `assert(false)` is the appropriate response in both cases.

Reply:
I really disagree on this point. We have had many examples over the years of asserts that people have added to our networking code for reasons that seem justifiable (along the lines of the reasons you've given), only to discover later that (usually due to an unexpected combination of code changes elsewhere) we ended up opening up a remote crash bug in deployed software that can take down the network -- in situations where we could easily have written code more defensively so that our software wouldn't crash.

We have Assume() exactly to avoid situations where a crash is worse than trying to gracefully recover, and we should err on the side of using that, particularly in networking code that could be triggered by an adversary against the whole network.